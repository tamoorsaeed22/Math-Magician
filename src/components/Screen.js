import PropTypes from 'prop-types';
import React from 'react';

export default function Screen({ value }) {
  const { total, next, operation } = value;
  return (
    <div className="output">
      <div className="current-operand" data-testid="output">
        {total}
        {operation}
        {next}
      </div>
    </div>
  );
}

Screen.propTypes = {
  value: PropTypes.string.isRequired,
};
