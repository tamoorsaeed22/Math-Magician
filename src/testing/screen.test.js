import renderer from 'react-test-renderer';
import Screen from '../components/Screen';

const obj = { total: null, next: '5', operation: null };

it('renders homepage correctly', () => {
  const tree = renderer.create(
    <Screen value={obj.next} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
