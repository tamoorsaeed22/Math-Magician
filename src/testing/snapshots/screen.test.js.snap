// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage correctly 1`] = `
<div
  className="output"
>
  <div
    className="current-operand"
    data-testid="output"
  />
</div>
`;